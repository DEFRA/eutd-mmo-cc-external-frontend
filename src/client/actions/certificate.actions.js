import { isError, showFullPageError } from './index';

export const GET_SUMMARY_DOCUMENT = 'GET_SUMMARY_DOCUMENT';
export const SHOW_SUMMARY_DOCUMENT_ERRORS = 'SHOW_SUMMARY_DOCUMENT_ERRORS';
export const GET_SUMMARY_UNAUTHORISED = 'GET_SUMMARY_DOCUMENT_UNAUTHORISED';

export function showSummaryCertificateErrors(data) {
  return { type: SHOW_SUMMARY_DOCUMENT_ERRORS, payload: { validationErrors: data } };
}

export const getSummaryCertificate = (journey, documentNumber = undefined) => async (dispatch, getState, { orchestrationApi }) => {
  const config = (documentNumber)
    ? { headers: { documentnumber: documentNumber } }
    : {};

  await orchestrationApi.get(`/certificate/${journey}`, config)
    .then(
      res => {
        dispatch({ type: GET_SUMMARY_DOCUMENT, payload: res.data });
      },
      error => {
        if (isError(error)) {
          dispatch(showFullPageError());
        } else {
          dispatch(unauthorised());
        }
      });

    function unauthorised() { return { type: GET_SUMMARY_UNAUTHORISED }; }
};