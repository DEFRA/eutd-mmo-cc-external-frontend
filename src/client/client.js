// Startup point for the client side application
import 'babel-polyfill';
import React from 'react';
import ReactDOM from 'react-dom';
import {Router, withRouter} from 'react-router-dom';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { renderRoutes } from 'react-router-config';
import axios from 'axios';
import Routes from './Routes';
import reducers from './reducers';
import { createBrowserHistory } from 'history';

const composeEnhancers = process.env.NODE_ENV === 'development' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__  ?
  window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : compose;
  

const paramSerializerFn = params => {
  let output = '';
  Object.keys(params).forEach((k) => {
    output+= `${k}=${params[k]}&`;
  });
  return encodeURI(output);
};

const axiosResponseInterceptorSuccessFn = response => response;
const axiosResponseInterceptorErrorFn = error => {
  if (error.response.status === 599) {
    // Let's check if this is because we're logged out
    const empPromise = new Promise(() => {});
    const searchParam = error.request.responseURL.slice(error.request.responseURL.search(/[?]/g));
    let searchParams = new URLSearchParams(searchParam);
    if(searchParams.get('notLoggedInErr') === 'yes') {
      window.location = '/logout?backToPath=/server-logout';
      return empPromise;
    }
    // It's always a timeout...
    window.location = '/timed-out';
    return new Promise(() => {});
  } else {
    return new Promise((_,rej) => rej(error));
  }
};

const axiosInstanceForOrchService = axios.create({
  baseURL: '/orchestration/api/v1/',
  paramsSerializer: paramSerializerFn,
  timeout: window.INITIAL_STATE.config.axiosTimeout
});

axiosInstanceForOrchService.interceptors.response.use(axiosResponseInterceptorSuccessFn, axiosResponseInterceptorErrorFn);

const axiosInstanceForRefService = axios.create({
  baseURL: '/reference/api/v1/',
  paramsSerializer: paramSerializerFn,
  timeout: window.INITIAL_STATE.config.axiosTimeout
});

axiosInstanceForRefService.interceptors.response.use(axiosResponseInterceptorSuccessFn, axiosResponseInterceptorErrorFn);

const axiosInstanceForDynamixService = axios.create({
  baseURL: '/dynamix/',
  paramsSerializer: paramSerializerFn,
  timeout: window.INITIAL_STATE.config.axiosTimeout
});

axiosInstanceForDynamixService.interceptors.response.use(axiosResponseInterceptorSuccessFn, axiosResponseInterceptorErrorFn);

const store = createStore(
  reducers,
  window.INITIAL_STATE,
  composeEnhancers(applyMiddleware(
    // The difference between orch/ref service api is not significant as they both go through proxy
    // However it is significant in the server side store. To stick with same action creators we're
    // passing down same axios instance to both orchestration and reference service api .
    thunk.withExtraArgument( {
      orchestrationApi: axiosInstanceForOrchService,
      referenceServiceApi: axiosInstanceForRefService,
      dynamixApi: axiosInstanceForDynamixService } )
  ))
);

const ScrollToTop = withRouter( class ScrollToTop extends React.Component {
  componentDidMount() {
  }

  render() {
    return this.props.children;
  }
});

const history = createBrowserHistory();


ReactDOM.hydrate(
  <Provider store={store}>
    <Router history={history}>
      <ScrollToTop>
        <div>{renderRoutes(Routes)}</div>
      </ScrollToTop>
    </Router>
  </Provider>,
  document.querySelector('#root')
);
