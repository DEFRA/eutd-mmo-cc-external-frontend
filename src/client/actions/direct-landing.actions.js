import { beginApiCall, apiCallFailed, isForbidden, showFullPageError, CLEAR_ERRORS, isSilverLineError } from './index';
import { exportPayloadActionTypes } from '../actions';


export const GET_DIRECT_LANDING_PRODUCTS = 'get_direct_landing_products';
export const CLEAR_DIRECT_LANDING = 'clear_direct_landing';
export const DIRECT_LANDING_PRODUCTS_FAILED = 'direct_landing_products_failed';
export const POST_DIRECT_LANDING_PRODUCTS_UNAUTHORISED = 'post_direct_landing_products_unauthorised';
export const GET_DIRECT_LANDING_PRODUCTS_UNAUTHORISED = 'get_direct_landing_products_unauthorised';
export const POST_DIRECT_LANDING_PRODUCTS_WAF_ERROR = 'post_direct_landing_prodcts_waf_error';

export const getDirectLandings =
  (documentNumber) =>
  async (dispatch, _getState, { orchestrationApi }) => {
    const config = { headers: { documentnumber: documentNumber } };

    try {
      dispatch(beginApiCall());

      const res = await orchestrationApi.get(
        '/export-certificates/export-payload/direct-landings',
        config
      );

      await dispatch({ type: GET_DIRECT_LANDING_PRODUCTS, payload: res.data });
    } catch (e) {
      if (isForbidden(e)) {
        dispatch({ type: GET_DIRECT_LANDING_PRODUCTS_UNAUTHORISED });
      } else {
        dispatch(apiCallFailed(e.response));
        throw new Error(
          `An error has occurred ${e.response.status} (${e.response.statusText})`
        );
      }
    }
  };

export const clearDirectLanding = () => ({ type: CLEAR_DIRECT_LANDING });

export const upsertDirectLanding = (landing, documentNumber = undefined) => (dispatch, getState, { orchestrationApi } ) => {
  const config = (documentNumber)
    ? {
        headers: {
          documentnumber: documentNumber
        }
    }
    : {};

  return orchestrationApi.post('/export-certificates/direct-landing/validate', landing, config)
    .then(
      res => {
        dispatch(landingUpserted(res.data));
        dispatch(clearErrors());
      },
      error => {
        if (error.response.status === 400) {
          dispatch(landingUpserted(error.response.data));
          throw new Error();
        } else if (isSilverLineError(error)) {
          dispatch({ type: POST_DIRECT_LANDING_PRODUCTS_WAF_ERROR, supportID: error.response.data.match(/\d+/)[0] });
        } else if (isForbidden(error)) {
          dispatch({ type: POST_DIRECT_LANDING_PRODUCTS_UNAUTHORISED });
        } else {
          dispatch(showFullPageError());
        }
      });

  function landingUpserted(exportPayload) { return { type: exportPayloadActionTypes.LANDING_UPSERTED, exportPayload, errors: exportPayload.errors }; }
  function clearErrors() { return { type: CLEAR_ERRORS }; }
};
