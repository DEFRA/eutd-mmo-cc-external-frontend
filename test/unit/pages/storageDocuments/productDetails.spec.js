import {  mount } from 'enzyme';
import * as React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import { render, screen } from '@testing-library/react';
import { component as productDetails } from '../../../../src/client/pages/storageNotes/productDetails';
import ProductDetails from '../../../../src/client/pages/storageNotes/productDetails';
import { getStorageNotesFromRedis, clearStorageNotes } from '../../../../src/client/actions';
import SUT from '../../../../src/client/pages/storageNotes/productDetails';

jest.mock('../../../../src/client/actions');

describe('loadData', () => {

  const store = {
    dispatch: () => {
      return new Promise((res) => {
        res();
      });
    }
  };

  const journey = 'storageDocument';
  const documentNumber = 'some-document-number';

  beforeEach(() => {
    getStorageNotesFromRedis.mockReturnValue({ type: 'GET_STORAGE_NOTES_FROM_REDIS' });
  });

  it('will call all methods needed to load the component', async () => {
    SUT.documentNumber = documentNumber;

    await SUT.loadData(store, journey);

    expect(getStorageNotesFromRedis).toHaveBeenCalledWith(documentNumber);
  });

  it('should redirect to the forbidden page if the storage notes is unauthorised', () => {    
      const mockPush = jest.fn();
      new productDetails.WrappedComponent({
      match: {
        params: {
          documentNumber: ''
        }
      },
      storageNotes: { unauthorised: true} ,
      history: {
        push: mockPush
      },
      route: {
        previousUri: ':documentNumber/some-previous-page'
      }
  }).componentDidUpdate();

  expect(mockPush).toHaveBeenCalledWith('/forbidden');
  
  });
});

describe('Back to progress page link', () => {
  let wrapper;
  beforeEach(() => {
    clearStorageNotes.mockReturnValue({ type: 'CLEAR_STORAGE_NOTES' });

    const mockStore = configureStore([thunk]);

    const props = {
      match: {
        params: {
          documentNumber: 'GBR-2021-CC-51F7BCC0A',
        },
      },
      route: {
        path: '/create-storage-document/123456789/add-product-to-this-consignment',
        previousUri:
          '/create-storage-document/:documentNumber/add-product-to-this-consignment',
        journey: 'storageNotes',
        progressUri: '/create-storage-document/progress',
        saveAsDraftUri: '/create-storage-document/storage-documents',
        title: 'Create a UK storage document - GOV.UK',
      }
    };

    const store = mockStore({
      clear: clearStorageNotes,
      global: {
        allFish: [],
      },
      errors: {},
      index: 0,
      scientificName: '',
      selectedDateOfUnloading: undefined,
      species: '',
      storageNotes: {
        addAnotherProduct: 'No',
        addAnotherStorageFacility: 'No',
        catches: [],
        errors: {},
        errorsUrl: '',
        storageFacilities: [],
        validationErrors: [{}],
      }
    });
    wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ProductDetails.component {...props} />
        </MemoryRouter>
      </Provider>
    );
  });

  it('should have a back to the progress page link', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper.find('a[href="/create-storage-document/progress"]').exists()).toBeTruthy();
  });
});

describe('Product details snapshot', () => {
  let wrapper;
  beforeEach(() => {
    clearStorageNotes.mockReturnValue({ type: 'CLEAR_STORAGE_NOTES' });

    const mockStore = configureStore([thunk]);

    const props = {
      match: {
        params: {
          documentNumber: 'GBR-2021-CC-51F7BCC0A',
        },
      },
      history: {
        push: jest.fn(),

        location: {
          pathname:
            '/create-storage-document/123456789/add-product-to-this-consignment/0',
          search: '',
          state: undefined,
        },
      },
      route: {
        path: '/create-storage-document/123456789/add-product-to-this-consignment/0',
        previousUri:
          '/create-storage-document/:documentNumber/add-product-to-this-consignment/0',
        progressUri: '/create-storage-document/:documentNumber/progress',
        journey: 'storageNotes',
        saveAsDraftUri: '/create-storage-document/storage-documents',
        title: 'Create a UK storage document - GOV.UK',
      },
      storageNotes: {
        addAnotherProduct: 'No',
        addAnotherStorageFacility: 'No',
        catches: [
          {
            certificateNumber: 'ABCD',
            commodityCode: '123456',
            dateOfUnloading: '14/10/2021',
            id: 'abcd-1634218473',
            placeOfUnloading: 'Dover',
            product: 'Atlantic cod (COD)',
            productWeight: '100',
            scientificName: '',
            transportUnloadedFrom: 'ABCD1234',
            weightOnCC: '100',
          },
        ],
        errors: {},
        errorsUrl: '',
        storageFacilities: [
          {
            _facilityUpdated: false,
            facilityAddressOne: 'MMO, LANCASTER HOUSE, HAMPSHIRE COURT',
            facilityBuildingName: 'LANCASTER HOUSE',
            facilityBuildingNumber: '',
            facilityCountry: 'ENGLAND',
            facilityCounty: 'TYNESIDE',
            facilityName: 'Some facility',
            facilityPostcode: 'NE4 7YH',
            facilityStreetName: 'HAMPSHIRE COURT',
            facilitySubBuildingName: 'MMO',
            facilityTownCity: 'NEWCASTLE UPON TYNE',
          },
        ],
        validationErrors: [{}],
      },
      clear: clearStorageNotes
    };

    const store = mockStore({
      clear: clearStorageNotes,
      global: {
        allFish: [],
      },
      errors: {},
      index: 0,
      scientificName: '',
      selectedDateOfUnloading: undefined,
      species: '',
      history: {
        push: jest.fn(),

        location: {
          pathname:
            '/create-storage-document/123456789/add-product-to-this-consignment/0',
          search: '',
          state: undefined,
        },
      },
      storageNotes: {
        addAnotherProduct: 'No',
        addAnotherStorageFacility: 'No',
        catches: [
          {
            certificateNumber: 'ABCD',
            commodityCode: '123456',
            dateOfUnloading: '14/10/2021',
            id: 'abcd-1634218473',
            placeOfUnloading: 'Dover',
            product: 'Atlantic cod (COD)',
            productWeight: '100',
            scientificName: '',
            transportUnloadedFrom: 'ABCD1234',
            weightOnCC: '100',
          },
        ],
        errors: {},
        errorsUrl: '',
        storageFacilities: [
          {
            _facilityUpdated: false,
            facilityAddressOne: 'MMO, LANCASTER HOUSE, HAMPSHIRE COURT',
            facilityBuildingName: 'LANCASTER HOUSE',
            facilityBuildingNumber: '',
            facilityCountry: 'ENGLAND',
            facilityCounty: 'TYNESIDE',
            facilityName: 'Some facility',
            facilityPostcode: 'NE4 7YH',
            facilityStreetName: 'HAMPSHIRE COURT',
            facilitySubBuildingName: 'MMO',
            facilityTownCity: 'NEWCASTLE UPON TYNE',
          },
        ],
        validationErrors: [{}],
      },
    });
    wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ProductDetails.component {...props} />
        </MemoryRouter>
      </Provider>
    );
  });

  it('should render the product details component and take snapshot', () => {
    expect(wrapper).toBeDefined();
    const { container } = render(wrapper);
    expect(container).toMatchSnapshot();
  });
});

describe('Product details errors', () => {
  let wrapper;
  const documentNumber = 'GBR-2021-CC-51F7BCC0A';
  beforeEach(() => {
    clearStorageNotes.mockReturnValue({ type: 'CLEAR_STORAGE_NOTES' });

    const mockStore = configureStore([thunk]);

    const storageNotes = {
      addAnotherProduct: 'No',
      addAnotherStorageFacility: 'No',
      catches: [{}],
      errors: {
        'catches-0-product': 'sdAddProductToConsignmentProductNameErrorNull',
        'catches-0-certificateNumber':
          'sdAddProductToConsignmentCertificateNumberErrorNull',
        'catches-0-commodityCode':
          'sdAddProductToConsignmentCommodityCodeErrorNull',
        'catches-0-dateOfUnloading':
          'sdAddProductToConsignmentDateOfUnloadingErrorNull',
        'catches-0-placeOfUnloading':
          'sdAddProductToConsignmentPlaceOfUnloadingErrorNull',
        'catches-0-productWeight':
          'sdAddProductToConsignmentExportWeightErrorNull',
        'catches-0-transportUnloadedFrom':
          'sdAddProductToConsignmentTransportDetailsErrorNull',
        'catches-0-weightOnCC': 'sdAddProductToConsignmentWeightOnCCErrorNull',
      },
      errorsUrl: '/',
      storageFacilities: [
        {
          _facilityUpdated: false,
          facilityAddressOne: 'MMO, LANCASTER HOUSE, HAMPSHIRE COURT',
          facilityBuildingName: 'LANCASTER HOUSE',
          facilityBuildingNumber: '',
          facilityCountry: 'ENGLAND',
          facilityCounty: 'TYNESIDE',
          facilityName: 'Some facility',
          facilityPostcode: 'NE4 7YH',
          facilityStreetName: 'HAMPSHIRE COURT',
          facilitySubBuildingName: 'MMO',
          facilityTownCity: 'NEWCASTLE UPON TYNE',
        },
      ],
      validationErrors: [{}],
    };

    const props = {
      match: {
        params: {
          documentNumber,
        },
      },
      history: {
        push: jest.fn(),

        location: {
          pathname: `/create-storage-document/${documentNumber}/add-product-to-this-consignment`,
          search: '',
          state: undefined,
        },
      },
      route: {
        journey: 'storageNotes',
        title: 'Create a UK storage document - GOV.UK',
        path: `/create-storage-document/${documentNumber}/add-product-to-this-consignment`,
        previousUri: `/create-storage-document/${documentNumber}/add-product-to-this-consignment`,
        saveAsDraftUri: `/create-storage-document/${documentNumber}/you-have-added-a-product`,
        nextUri: `/create-storage-document/${documentNumber}/you-have-added-a-product`,
        progressUri: `/create-storage-document/${documentNumber}/progress`
      },
      storageNotes,
      clear: clearStorageNotes,
    };

    const store = mockStore({
      clear: clearStorageNotes,
      global: {
        allFish: [],
      },
      index: 0,
      scientificName: '',
      selectedDateOfUnloading: undefined,
      species: '',
      history: {
        push: jest.fn(),

        location: {
          pathname:
            '/create-storage-document/123456789/add-product-to-this-consignment',
          search: '',
          state: undefined,
        },
      },
      storageNotes,
    });
    wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <ProductDetails.component {...props} />
        </MemoryRouter>
      </Provider>
    );
  });

  it('should render the product details component with errors', () => {
    render(wrapper);

    expect(screen.getByText('Enter the FAO code or product name')).toBeTruthy();
    expect(screen.getByText('Enter the document number')).toBeTruthy();
    expect(
      screen.getByText(
        'Commodity code must be between 6 and 12 characters, and contain only letters, numbers, full stops, colons, hyphens, and slashes'
      )
    ).toBeTruthy();
    expect(
      screen.getByText('Enter the date product entered the UK')
    ).toBeTruthy();
    expect(
      screen.getByText('Enter the place product entered the UK')
    ).toBeTruthy();
    expect(screen.getByText('Enter the export weight in kg')).toBeTruthy();
  });
});