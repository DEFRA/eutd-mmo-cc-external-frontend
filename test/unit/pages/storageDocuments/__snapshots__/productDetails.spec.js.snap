// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product details snapshot should render the product details component and take snapshot 1`] = `
<div>
  <div
    class="css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <a
            class="e1vcssft0 css-zk98nb-Anchor-StyledHoc e4u1u2g0"
            href="/orchestration/api/v1/processingStatement/back?n=/create-storage-document/undefined/add-product-to-this-consignment/0"
          >
            Back
          </a>
          <h1
            class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
          >
            Add a product to this consignment
          </h1>
        </div>
      </div>
      <form
        action="/orchestration/api/v1/storageNotes/saveAndValidate?c=/create-storage-document/undefined/add-product-to-this-consignment/0&n=/create-storage-document/undefined/you-have-added-a-product"
        method="POST"
      >
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <div
              class="input-species-ps properties-group autocomplete-sd "
            >
              <label
                class="e1vcssft0 css-137pbp-StyledLabel-StyledHoc e4u1u2g0"
                for="catches.0.product"
                id="label-catches-0-product"
                name="catches-0-product"
              >
                <div
                  id="speciesLabel"
                >
                  FAO code or product name
                </div>
                <span
                  class="e1vcssft0 css-1whl0n4-StyledHint-StyledHoc e4u1u2g0"
                >
                  Enter part of the FAO code or product name for suggestions e.g. Atlantic Cod (COD). You must choose from the suggestions provided to avoid invalid documents.
                </span>
                <div
                  class="autocomplete__wrapper"
                >
                  <div
                    style="border: 0px; height: 1px; margin-bottom: -1px; margin-right: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 1px;"
                  >
                    <div
                      aria-atomic="true"
                      aria-live="polite"
                      id="catches.0.product__status--A"
                      role="status"
                    />
                    <div
                      aria-atomic="true"
                      aria-live="polite"
                      id="catches.0.product__status--B"
                      role="status"
                    />
                  </div>
                  
                  <input
                    aria-activedescendant="false"
                    aria-autocomplete="list"
                    aria-describedby="catches.0.product__assistiveHint"
                    aria-expanded="false"
                    aria-owns="catches.0.product__listbox"
                    autocomplete="off"
                    class="autocomplete__input autocomplete__input--default"
                    id="catches.0.product"
                    name="catches.0.product"
                    placeholder=""
                    role="combobox"
                    type="text"
                    value="Atlantic cod (COD)"
                  />
                  <ul
                    class="autocomplete__menu autocomplete__menu--overlay autocomplete__menu--hidden"
                    id="catches.0.product__listbox"
                    role="listbox"
                  >
                    <li
                      aria-posinset="1"
                      aria-selected="false"
                      aria-setsize="1"
                      class="autocomplete__option"
                      id="catches.0.product__option--0"
                      role="option"
                      tabindex="-1"
                    >
                      Atlantic cod (COD)
                    </li>
                  </ul>
                  <span
                    id="catches.0.product__assistiveHint"
                    style="display: none;"
                  >
                    When autocomplete results are available use up and down arrows to review and enter to select.  Touch device users, explore by touch or with swipe gestures.
                  </span>
                </div>
              </label>
              <input
                name="productCode"
                type="hidden"
                value=""
              />
            </div>
            <br />
            <details>
              <summary
                class="govuk-details__summary"
              >
                I cannot find the species
              </summary>
              <div
                class="govuk-details__text "
              >
                For best results, search for the common English name or the FAO code (if known) as species nicknames are not supported.
                <p
                  class="css-sq4ju6"
                >
                  Some species are exempt
                  :Â 
                  <a
                    aria-label="Link opens in a new window"
                    href="https://eur-lex.europa.eu/LexUriServ/LexUriServ.do?uri=OJ:L:2011:057:0010:0018:EN:PDF"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Species exempt from Storage Documents (europa.eu)
                     
                    <span
                      class="govuk-visually-hidden"
                    >
                      (opens in new tab)
                    </span>
                  </a>
                </p>
                If you cannot find the species and it is not exempt, call 
                0330 159 1989.
              </div>
            </details>
            <br />
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-n58wva-StyledColumn e4u1u2g0"
          >
            <label
              class="e1vcssft0 e1vcssft0 css-5tg97n-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
              for="catches.0.commodityCode"
              id="catches-0-commodityCode"
              style="margin-top: -1em;"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              >
                Commodity code
              </span>
              
              <input
                autocomplete="off"
                class="formControl mediumInput e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
                id="catches.0.commodityCode"
                name="catches.0.commodityCode"
                type="text"
                value="123456"
              />
            </label>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-n58wva-StyledColumn e4u1u2g0"
          >
            <label
              class="e1vcssft0 e1vcssft0 css-5tg97n-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
              for="catches.0.productWeight"
              id="catches-0-productWeight"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              >
                Export weight in kg
              </span>
              
              <input
                autocomplete="off"
                class="formControl smallInput e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
                id="catches.0.productWeight"
                name="catches.0.productWeight"
                type="text"
                value="100"
              />
            </label>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-1w80l5h-StyledColumn e4u1u2g0"
          >
            <h1
              class="e1vcssft0 css-11nzzut-StyledHeader-StyledHoc e4u1u2g0"
            >
              UK entry document
            </h1>
            <p
              style="color: rgb(111, 119, 123);"
            >
              This is the last document used for the product to enter the UK. It could be a 3rd country catch certificate, processing statement or storage document.
            </p>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-n58wva-StyledColumn e4u1u2g0"
          >
            <div
              class="date-field-with-picker"
              id="catches-0-dateOfUnloading"
            >
              <div
                class="column-field-picker"
              >
                <div
                  class="date-field e1vcssft0 css-vav5a4-StyledContainer-StyledHoc e4u1u2g0"
                >
                  <span
                    class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
                  >
                    <span
                      class="label-product-details-form e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
                      id="date-field-label-text"
                    >
                      Date product entered the UK
                    </span>
                  </span>
                  <span
                    class="e1vcssft0 css-1whl0n4-StyledHint-StyledHoc e4u1u2g0"
                  >
                    For example, 31 03 1980
                  </span>
                  
                  <div
                    class="css-bw34j1-StyledList e107rt5v2"
                  >
                    <label
                      class="e107rt5v1 css-iwpl6b-StyledLabel-StyledHoc-StyledLabel e4u1u2g0"
                    >
                      <span
                        class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
                      >
                        Day
                      </span>
                      <input
                        class="css-lktfuz-StyledInput e107rt5v0"
                        name="dayInputName"
                        type="number"
                        value="14"
                      />
                    </label>
                    <label
                      class="e107rt5v1 css-iwpl6b-StyledLabel-StyledHoc-StyledLabel e4u1u2g0"
                    >
                      <span
                        class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
                      >
                        Month
                      </span>
                      <input
                        class="css-lktfuz-StyledInput e107rt5v0"
                        name="monthInputName"
                        type="number"
                        value="10"
                      />
                    </label>
                    <label
                      class="e107rt5v1 css-17bpbza-StyledLabel-StyledHoc-StyledLabel e4u1u2g0"
                    >
                      <span
                        class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
                      >
                        Year
                      </span>
                      <input
                        class="css-lktfuz-StyledInput e107rt5v0"
                        name="yearInputName"
                        type="number"
                        value="2021"
                      />
                    </label>
                  </div>
                </div>
                <div
                  class="date-picker "
                >
                  <button>
                    <img
                      alt="calendar"
                      src="/static/assets/images/calendar.png"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-n58wva-StyledColumn e4u1u2g0"
          >
            <label
              class="e1vcssft0 e1vcssft0 css-5tg97n-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
              for="catches.0.placeOfUnloading"
              id="catches-0-placeOfUnloading"
              style="margin-top: -2em;"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              >
                Port where product entered the UK
              </span>
              <span
                class="e1vcssft0 css-1whl0n4-StyledHint-StyledHoc e4u1u2g0"
              >
                For example, Felixstowe or Dover.
              </span>
              
              <input
                autocomplete="off"
                class="formControl mediumInput e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
                id="catches.0.placeOfUnloading"
                name="catches.0.placeOfUnloading"
                type="text"
                value="Dover"
              />
            </label>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-n58wva-StyledColumn e4u1u2g0"
          >
            <label
              class="e1vcssft0 e1vcssft0 css-5tg97n-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
              for="catches.0.transportUnloadedFrom"
              id="catches-0-transportUnloadedFrom"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              >
                Transport details
              </span>
              <span
                class="e1vcssft0 css-1whl0n4-StyledHint-StyledHoc e4u1u2g0"
              >
                For example, the name of a container ship, flight number or vehicle registration.
              </span>
              
              <input
                autocomplete="off"
                class="formControl e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
                id="catches.0.transportUnloadedFrom"
                name="catches.0.transportUnloadedFrom"
                type="text"
                value="ABCD1234"
              />
            </label>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-n58wva-StyledColumn e4u1u2g0"
          >
            <label
              class="e1vcssft0 e1vcssft0 css-5tg97n-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
              for="catches.0.certificateNumber"
              id="catches-0-certificateNumber"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              >
                Document number
              </span>
              <span
                class="e1vcssft0 css-1whl0n4-StyledHint-StyledHoc e4u1u2g0"
              >
                Enter 1 catch certificate document number per line.
              </span>
              
              <input
                autocomplete="off"
                class="formControl largeInput e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
                id="catches.0.certificateNumber"
                maxlength="54"
                name="catches.0.certificateNumber"
                type="text"
                value="ABCD"
              />
            </label>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <div
            class="css-n58wva-StyledColumn e4u1u2g0"
          >
            <label
              class="e1vcssft0 e1vcssft0 css-5tg97n-StyledLabel-StyledHoc-StyledHoc e4u1u2g0"
              for="catches.0.weightOnCC"
              id="catches-0-weightOnCC"
            >
              <span
                class="e1vcssft0 css-3e01s4-StyledLabelText-StyledHoc e4u1u2g0"
              >
                Weight on document in kg
              </span>
              
              <input
                autocomplete="off"
                class="formControl smallInput e1vcssft0 css-4n7f6s-StyledInput-StyledHoc e4u1u2g0"
                id="catches.0.weightOnCC"
                name="catches.0.weightOnCC"
                type="text"
                value="100"
              />
            </label>
          </div>
        </div>
        <div
          class="css-1lzr9ic-StyledRow e4u1u2g0"
        >
          <button
            class="css-17uw1q2-StyledButton-StyledHoc e4u1u2g0"
            formaction="/orchestration/api/v1/storageNotes/saveAndValidate?saveAsDraftUrl=/create-storage-document/storage-documents&c=/create-storage-document/add-product-to-this-consignment/0&n=/create-storage-document/you-have-added-a-product"
            id="saveAsDraft"
            name="saveAsDraft"
            type="submit"
          >
            Save as Draft
          </button>
          <input
            name="journey"
            type="hidden"
            value="storageNotes"
          />
          <input
            name="currentUri"
            type="hidden"
            value="/create-storage-document/123456789/add-product-to-this-consignment/0"
          />
          <button
            class="css-1gc9650-StyledButton-StyledHoc e4u1u2g0"
            id="continue"
          >
            Save and continue
          </button>
        </div>
      </form>
      <a
        href="/create-storage-document/undefined/progress"
        id="backToProgress"
      >
        Back to Your Progress
        <span
          class="govuk-visually-hidden"
        >
          back to your progress page link
        </span>
      </a>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <hr
            class="help-link"
          />
          <h2
            class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
          >
            Need help?
          </h2>
          <a
            href="undefined"
            rel="noopener noreferrer"
            target="_blank"
          >
            Get help exporting fish from the UK (gov.uk)
             
            <span
              class="govuk-visually-hidden"
            >
              (opens in new tab)
            </span>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
